src/distribution.cpp

```diff
@@ -287,6 +287,42 @@ double Equiprobable::sample(uint64_t* seed) const
   return xl + ((n - 1) * r - i) * (xr - xl);
 }
 
+//==============================================================================
+// Mixture implementation
+//==============================================================================
+
+Mixture::Mixture(pugi::xml_node node)
+{
+  // Read and initialize tabular distribution
+  //auto params = get_node_array<double>(node, "parameters");
+
+  double cumsum = 0.0;
+  for (pugi::xml_node pair = node.child("pair"); pair; pair = pair.next_sibling("pair")) {
+    if (!pair.attribute("probability")) openmc::fatal_error("Mixture pair element does not have probability.");
+    if (!pair.child("dist")) openmc::fatal_error("Mixture pair element does not have a distribution.");
+    double p = std::stod(pair.attribute("probability").value());
+    p_.push_back(p);
+    cumsum += p;
+    c_.push_back(cumsum);
```

Comment:
Right now there's no guarantee that the probabilities sum to one. We should probably renormalize the probabilities first so that they _do_ sum to one. On the Python side, we might want to do the same thing when exporting to ensure that the probabilities in XML sum to one in the first place.